File: NCAA/scripts/_check_big_dogs.mjs
#!/usr/bin/env node
import fs from "fs";

const store = JSON.parse(fs.readFileSync("data/history.json", "utf8"));
const runs = store.runs || [];

// Check: for 7+ point dogs with pCover >= 0.60, show the actual games
let wins = 0, losses = 0;
const lossList = [];

for (const r of runs) {
  for (const g of r.games) {
    if (g.homeScore === undefined || g.awayScore === undefined) continue;
    if (g.status === "SKIPPED" || g.status === "MISSING_STATS") continue;
    const absLine = Math.abs(g.line);
    if (absLine < 7) continue;

    const homeDog = g.line > 0;
    if (!homeDog && g.line >= 0) continue; // line=0, skip
    const awayDog = g.line < 0;

    const dogSide = homeDog ? "home" : "away";
    const dogP = homeDog ? (g.pHomeCover || 0) : (g.pAwayCover || 0);
    if (dogP < 0.60) continue;

    const margin = g.homeScore - g.awayScore;
    const v = margin + g.line;
    const covered = dogSide === "home" ? v > 0 : v < 0;

    if (covered) {
      wins++;
    } else {
      losses++;
      lossList.push({
        date: r.date,
        game: `${g.away} @ ${g.home}`,
        line: g.line,
        score: `${g.awayScore}-${g.homeScore}`,
        dogP: dogP.toFixed(3),
        margin: margin,
        spreadResult: v
      });
    }
  }
}

console.log(`7+ point dogs with pCover >= 0.60: ${wins}-${losses}`);
console.log(`\nLosses:`);
for (const l of lossList) {
  console.log(`  ${l.date} | ${l.game} | line: ${l.line} | score: ${l.score} | dogP: ${l.dogP} | margin: ${l.margin} | spread result: ${l.spreadResult}`);
}

// Now check: what % of ALL 7+ point dogs cover (regardless of model)?
let allBigDogW = 0, allBigDogL = 0;
for (const r of runs) {
  for (const g of r.games) {
    if (g.homeScore === undefined || g.awayScore === undefined) continue;
    if (g.status === "SKIPPED" || g.status === "MISSING_STATS") continue;
    const absLine = Math.abs(g.line);
    if (absLine < 7) continue;

    const homeDog = g.line > 0;
    if (!homeDog && g.line >= 0) continue;
    const dogSide = homeDog ? "home" : "away";

    const margin = g.homeScore - g.awayScore;
    const v = margin + g.line;
    const covered = dogSide === "home" ? v > 0 : v < 0;
    if (covered) allBigDogW++;
    else allBigDogL++;
  }
}

console.log(`\nALL 7+ point dogs (no pCover filter): ${allBigDogW}-${allBigDogL} (${(allBigDogW/(allBigDogW+allBigDogL)*100).toFixed(1)}%)`);

// Check a few sample games to verify the grading
console.log("\n=== SAMPLE 7+ pt dog wins (first 10) ===");
let shown = 0;
for (const r of runs) {
  if (shown >= 10) break;
  for (const g of r.games) {
    if (shown >= 10) break;
    if (g.homeScore === undefined || g.awayScore === undefined) continue;
    if (g.status === "SKIPPED" || g.status === "MISSING_STATS") continue;
    const absLine = Math.abs(g.line);
    if (absLine < 7) continue;
    const homeDog = g.line > 0;
    if (!homeDog && g.line >= 0) continue;
    const dogSide = homeDog ? "home" : "away";
    const dogP = homeDog ? (g.pHomeCover || 0) : (g.pAwayCover || 0);
    if (dogP < 0.60) continue;

    const margin = g.homeScore - g.awayScore;
    const v = margin + g.line;
    const covered = dogSide === "home" ? v > 0 : v < 0;
    if (!covered) continue;

    const dogTeam = dogSide === "home" ? g.home : g.away;
    const favTeam = dogSide === "home" ? g.away : g.home;
    console.log(`  ${r.date} | ${dogTeam} +${absLine} vs ${favTeam} | ${g.awayScore}-${g.homeScore} | dogP=${dogP.toFixed(3)} | spreadResult=${v.toFixed(1)}`);
    shown++;
  }
}

File: pyNCAA/scripts/_check_big_dogs.py
#!/usr/bin/env python3
# scripts/_check_big_dogs.py — Check 7+ point dogs with pCover >= 0.60
import json

store = json.loads(open("data/history.json").read())
runs = store.get("runs") or []

wins, losses = 0, 0
loss_list = []

for r in runs:
    for g in (r.get("games") or []):
        if g.get("homeScore") is None or g.get("awayScore") is None:
            continue
        if g.get("status") in ("SKIPPED", "MISSING_STATS"):
            continue
        abs_line = abs(g.get("line", 0))
        if abs_line < 7:
            continue
        home_dog = g.get("line", 0) > 0
        if not home_dog and g.get("line", 0) >= 0:
            continue
        dog_side = "home" if home_dog else "away"
        dog_p = g.get("pHomeCover", 0) if home_dog else g.get("pAwayCover", 0)
        if dog_p < 0.60:
            continue
        margin = g["homeScore"] - g["awayScore"]
        v = margin + g["line"]
        covered = v > 0 if dog_side == "home" else v < 0
        if covered:
            wins += 1
        else:
            losses += 1
            loss_list.append({
                "date": r.get("date"), "game": f"{g['away']} @ {g['home']}",
                "line": g["line"], "score": f"{g['awayScore']}-{g['homeScore']}",
                "dogP": f"{dog_p:.3f}", "margin": margin, "spreadResult": v
            })

print(f"7+ point dogs with pCover >= 0.60: {wins}-{losses}")
print("\nLosses:")
for ll in loss_list:
    print(f"  {ll['date']} | {ll['game']} | line: {ll['line']} | score: {ll['score']} | dogP: {ll['dogP']} | margin: {ll['margin']} | spread result: {ll['spreadResult']}")

# All 7+ point dogs (no pCover filter)
all_big_dog_w, all_big_dog_l = 0, 0
for r in runs:
    for g in (r.get("games") or []):
        if g.get("homeScore") is None or g.get("awayScore") is None:
            continue
        if g.get("status") in ("SKIPPED", "MISSING_STATS"):
            continue
        abs_line = abs(g.get("line", 0))
        if abs_line < 7:
            continue
        home_dog = g.get("line", 0) > 0
        if not home_dog and g.get("line", 0) >= 0:
            continue
        margin = g["homeScore"] - g["awayScore"]
        v = margin + g["line"]
        covered = (v > 0 if home_dog else v < 0)
        if covered:
            all_big_dog_w += 1
        else:
            all_big_dog_l += 1

print(f"\nALL 7+ point dogs (no pCover filter): {all_big_dog_w}-{all_big_dog_l} ({all_big_dog_w / (all_big_dog_w + all_big_dog_l) * 100:.1f}%)")

print("\n=== SAMPLE 7+ pt dog wins (first 10) ===")
shown = 0
for r in runs:
    if shown >= 10:
        break
    for g in (r.get("games") or []):
        if shown >= 10:
            break
        if g.get("homeScore") is None or g.get("awayScore") is None:
            continue
        if g.get("status") in ("SKIPPED", "MISSING_STATS"):
            continue
        abs_line = abs(g.get("line", 0))
        if abs_line < 7:
            continue
        home_dog = g.get("line", 0) > 0
        if not home_dog and g.get("line", 0) >= 0:
            continue
        dog_side = "home" if home_dog else "away"
        dog_p = g.get("pHomeCover", 0) if home_dog else g.get("pAwayCover", 0)
        if dog_p < 0.60:
            continue
        margin = g["homeScore"] - g["awayScore"]
        v = margin + g["line"]
        covered = v > 0 if dog_side == "home" else v < 0
        if not covered:
            continue
        dog_team = g["home"] if dog_side == "home" else g["away"]
        fav_team = g["away"] if dog_side == "home" else g["home"]
        print(f"  {r.get('date')} | {dog_team} +{abs_line} vs {fav_team} | {g['awayScore']}-{g['homeScore']} | dogP={dog_p:.3f} | spreadResult={v:.1f}")
        shown += 1
